server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/MySQLAffectedRows.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.oceanbase.odc.service.sqlcheck.rule;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.springframework.jdbc.core.JdbcOperations;
+
+import com.oceanbase.odc.core.shared.constant.DialectType;
+import com.oceanbase.odc.service.sqlcheck.SqlCheckContext;
+import com.oceanbase.odc.service.sqlcheck.SqlCheckRule;
+import com.oceanbase.odc.service.sqlcheck.SqlCheckUtil;
+import com.oceanbase.odc.service.sqlcheck.model.CheckViolation;
+import com.oceanbase.odc.service.sqlcheck.model.SqlCheckRuleType;
+import com.oceanbase.tools.sqlparser.statement.Statement;
+import com.oceanbase.tools.sqlparser.statement.delete.Delete;
+import com.oceanbase.tools.sqlparser.statement.insert.Insert;
+import com.oceanbase.tools.sqlparser.statement.update.Update;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * {@link MySQLAffectedRows}
+ *
+ * @author yiminpeng
+ * @version 1.0
+ * @date 2024-08-01 18:18
+ */
+@Slf4j
+public class MySQLAffectedRows implements SqlCheckRule {
+
+    private final Long maxSQLAffectedRows;
+
+    private final JdbcOperations jdbcOperations;
+
+    private final DialectType dialectType;
+
+    public MySQLAffectedRows(@NonNull Long maxSQLAffectedRows, DialectType dialectType,
+            JdbcOperations jdbcOperations) {
+        this.maxSQLAffectedRows = maxSQLAffectedRows <= 0 ? 0 : maxSQLAffectedRows;
+        this.jdbcOperations = jdbcOperations;
+        this.dialectType = dialectType;
+    }
+
+    /**
+     * Get the rule type
+     */
+    @Override
+    public SqlCheckRuleType getType() {
+        return SqlCheckRuleType.RESTRICT_SQL_AFFECTED_ROWS;
+    }
+
+    /**
+     * Execution rule check
+     */
+    @Override
+    public List<CheckViolation> check(@NonNull Statement statement, @NonNull SqlCheckContext context) {
+
+        if (statement instanceof Update || statement instanceof Delete || statement instanceof Insert) {
+            if (jdbcOperations == null) {
```

Comment:
don't throw the exception at the runtime, we should check it not null when we construct it~

Reply:
Ok, checked it before running.